src/explorer.ts

```diff
@@ -348,28 +371,17 @@ class KubernetesNamespaceResource extends KubernetesResource {
     }
 }
 
-class KubernetesServiceFolder extends KubernetesResourceFolder {
-    constructor() {
-        super(kuberesources.allKinds.service);
+class KubernetesPodSelectorFolder extends KubernetesResourceFolder {
+    constructor(readonly kind: kuberesources.ResourceKind, readonly fn: (kubectl: Kubectl) => Promise<kubectlUtils.PodSelector[]>) {
```

Comment:
I think I am also finding the name `PodSelector` confusing.  It implies some strategy or condition for selecting pods, whereas what I think it's used as is something like "a Kubernetes resource that _has_ a pod selector (that is, is associated with a set of pods."  Perhaps something like `PodSelectingResource` might help, though I fear this leads us towards a combinatorial explosion if we later need other traits too.

Would TypeScript intersection types help here, e.g. would modelling things as `KubernetesResource & HasPodSelector` (or `KubernetesResource & SelectsPods`) be clearer and more sustainable?

Reply:
Renamed, let me know what you think.